### PR TITLE
Chore do registration

### PR DESCRIPTION
**Pull request by Ange-Constance**
### **What does this PR do**
This pull request implementation of registering new drivers and operators.

### **Description**

1. The administrator should be able to register a new driver or a new operator.
2. An email containing a random password should be sent to the provider email.
3. The driver or operator should be able to sign in to the app using their email and the provided password.
4.Appropriate error messages are displayed on unsuccessful attempts.

### **How to manually test this**

1. Check in the branch chore-do-registration
2. Run the server
3. Open the Swagger API documentation to test driver & operators

### **Relevant Trello story**
[10-register-drivers-operators][(https://trello.com/c/SDfWF0QF/10-register-drivers-operators)](https://trello.com/c/SDfWF0QF/10-register-drivers-operators)

### **Screenshots** 

result
phantom

### CODE REVIEW

File: src/api/routes/signup.routes.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ *                 type: string
+ *               name:
+ *                 type: string
+ *             required:
+ *               - email
+ *               - name
+ *     responses:
+ *       200:
+ *         description: Successfully created operator account
+ *       400:
```
Comment (TuyizeeAnastase) [inline src/api/routes/signup.routes.js]: - I want only one endpoint
- add middleware to check if a user is not already registered by email
- our request body will be like this:
Screenshot from 2023-03-28 11-46-51

File: src/api/controllers/signup.controller.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  // Send an email to the provider email with the random password
+  // await sendEmail(email, 'Your password for the app', `Your password is ${password}`);
+
+  // Return a success message
+  res.status(StatusCodes.CREATED).json({
+    success: true,
+    message: "Driver registered successfully",
+    data: driverData,
+  });
```
Comment (TuyizeeAnastase) [inline src/api/controllers/signup.controller.js]: create one controller that will receive 
- will receive that body from the postman `name,email,role(operator or driver)`
- find role By RoleTitle `const role=findOne(req.body.role) `to get role `id`
- `const newUser={name,email,password:hashedPassword,role_id:role.id}` 
- finally save newUser

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]

**Review by thepatrickniyo — COMMENTED** [review]
There is a lint error causing the ci-cd action to fail

**Review by TuyizeeAnastase — APPROVED** [review]
Next time make sure, you are using naming conventions of PR descriptions refer to [this](https://github.com/atlp-rwanda/engineering-playbook/wiki)

**Pull request merged**